doc: various corrections and additions (#1)
DarkOceanInteractive/project-conventions: README.md

### Types of Naming Conventions

| Common Name   | Example                                        | Notes  |
| ------------- | ---------------------------------------------- | ------ |
| Pascal-case   | `DesertEagle`                                  |        |
| Camel-case    | `desertEagle`                                  |        |

> | Snake-case    | `desert_eagle`, `Desert_Eagle`, `desert_Eagle` |        |

Author: [Wikipedia](https://en.wikipedia.org/wiki/Snake_case) states that *"the first letter of each word written in lowercase".*

Reviewer: Sourcing Wikipedia gives me the shivers, but I admit my mistake.

There are many common rules for naming and maintaining **Classes** within
**Source Files**.

> - One source file should contain only one `public` class

Reviewer: ```suggestion
- Each source file should contain only one `public` class

```

    - `private`
- Finally, sort alphabetically.

Use [Pascal-case](#types-of-naming-conventions) for **Class Names**.

- Always prefix interface names with an `I`
- Always prefix abstract class names with an `A`

Use **Namespaces** to ensure your scoping of items won't conflict with existing
ones from other- or global namespaces. The project should at minimum use the
project name for its namespaces in order to prevent conflicts with imported
third party assets.

All **Public Functions** should have documentation in [the xmldoc format](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags),
containing at least a summary and a description of their parameters and return

> value if any.

Reviewer: ```suggestion
value, if any.
```

```csharp
/// <summary>
/// Fire a gun
/// </summary>
/// <param name="bulletType">Type of the bullet to fire</param>
/// <returns>True if the gun was able to fire, false if it wasn't for any reason</returns>
public bool Fire(BulletType bulletType)
{
  // Fire the gun.

> }

Author: If that is too much, we can shorten the example, I mainly wanted to talk about the `xmldoc` format in addition to the `<summary>`  tag.

Reviewer: I'm OK with this.

```

**Comments** should be used to describe intention, algorithmic overview, and/or
logical flow. The ideal comment allows someone other than the author to
understand a function's intended behavior and general operation.

- Comments should be placed on a separate line, above the code
- Comment text should begin with an uppercase letter
- Comment text should end with a period
- Insert one space between the comment delimiter and the comment text

Example:
```csharp
/// This is a comment.
```

> For **Spacing**, use a single space after a comma.

Reviewer 2: ```suggestion
For **Spacing**, use a single space after a comma between function arguments.
```

Example: `Console.In.Read(myChar, 0, 1);`

- Do not use a space after the parenthesis and function arguments
- Do not use spaces between a function name and parenthesis
- Do not use spaces inside brackets

> Class members should be accessed through the `this` keyword, to help tracking the

Reviewer: ```suggestion
Class members should be accessed through the `this` keyword, as this helps with tracking the

```

scope of each variable used.

Example:
```csharp
this.transform.position = this._startPosition;
```

### Compiling

All scripts should compile with **zero warnings** and **zero errors**. Warnings
and errors may quickly cascade into unexpected behavior.

### Variables

There are many common rules for naming and maintaining **Variables** within
**Classes**.

- Always use [Pascal-case](#types-of-naming-conventions) for public, internal,
  and protected variables
- Always use [Camel-case](#types-of-naming-conventions) for private, and local
  variables
- Always use [Pascal-case](#types-of-naming-conventions) for boolean, and enum
  variables.
- Always use [Pascal-case](#types-of-naming-conventions) for interfaces and
  classes
- Always prefix private variables with an underscore

> - Always prefix boolean variables with a verb such as `is`, `has`, ...

Reviewer: ```suggestion
- Always prefix boolean variables with a verb such as `is` or `has`.

```

- All variable names must be clear, unambiguous, and descriptive **nouns**
- Variable names must not be redundant with their context

> - Local variable and function parameter names should not conflict with other

Reviewer: ```suggestion
- Local variable- and function parameter names should not conflict with other

```

>   variables in the same scope (class variables, global variables, ...)

Reviewer: ```suggestion
  variables within the same scope (such as class variables, global variables, etc.)

```

- All **Serializable** variables should make use of value ranges if there is
  ever a value that a variable should not be set to

There are many common rules for naming and maintaining **Functions** and **Event
Handlers** within **Classes**.

- All function names should start with verbs
- All function names should be worded in present tense
- All function names should have context as to what they are doing
- All event handler names should start with `On`

> ### Design Principles

Author: Again that may be a bit much, but I feel the need to include them, at least not as hard rules that people **must** follow, but more as guidelines that people **should** follow as much as possible.

Reviewer: I think including design guidelines in the project conventions are not necessary, but a nice addition either way. I appreciate you adding them!

containing at least a summary and a description of their parameters and return
value, if any.
```csharp
/// <summary>
/// Fire a gun
/// </summary>
/// <param name="bulletType">Type of the bullet to fire</param>
/// <returns>True if the gun was able to fire, false if it wasn't for any reason</returns>
public bool Fire(BulletType bulletType)
{
  // Fire the gun.
}
```

**Comments** should be used to describe intention, algorithmic overview, and/or
logical flow. The ideal comment allows someone other than the author to
understand a function's intended behavior and general operation.

- Comments should be placed on a separate line, above the code
- Comment text should begin with an uppercase letter
- Comment text should end with a period
- Insert one space between the comment delimiter and the comment text

Example:
```csharp
/// This is a comment.
```

For **Spacing**, use a single space after a comma.

Example: `Console.In.Read(myChar, 0, 1);`

- Do not use a space after the parenthesis and function arguments
- Do not use spaces between a function name and parenthesis
- Do not use spaces inside brackets

> Write variables- and functions **modifiers** in that order:

Reviewer: ```suggestion
Variable- and function **modifiers** should be sorted in the following order:

```
